docs/flatcar-linux/kvm-libvirt.md

```diff
@@ -0,0 +1,297 @@
+```
+
+Build terraform-provider-libvirt from source because pool definition feature was not released yet:
+
+```sh
+$ go get github.com/dmacvicar/terraform-provider-libvirt
+$ cd $GOPATH/src/github.com/dmacvicar/terraform-provider-libvirt
+$ GO111MODULE=on GOFLAGS=-mod=vendor make install
+$ cp $GOPATH/bin/terraform-provider-libvirt ~/.terraform.d/plugins/terraform-provider-libvirt
+```
+
+Read [concepts](/docs/architecture/concepts.md) to learn about Terraform, modules, and organizing resources. Change to your infrastructure repository (e.g. `infra`).
+
+```sh
+cd infra/vmcluster
+```
+
+## Provider
+
+Create the `provider.tf` file:
+
+```tf
+provider "ct" {
+  version = "0.3.1"
+}
+
+provider "local" {
+  version = "~> 1.0"
+  alias   = "default"
+}
+
+provider "null" {
+  version = "~> 1.0"
+  alias   = "default"
+}
+
+provider "template" {
+  version = "~> 1.0"
+  alias   = "default"
+}
+
+provider "tls" {
+  version = "~> 1.0"
+  alias   = "default"
+}
+
+provider "libvirt" {
+  # TODO: add version once released
+  uri = "qemu:///system"
+  alias = "default"
+}
+```
+
+## Cluster
+
+Define a Kubernetes cluster using the module [kvm-libvirt/flatcar-linux/kubernetes](https://github.com/kinvolk/lokomotive-kubernetes/tree/master/kvm-libvirt/flatcar-linux/kubernetes).
+
+Create the `mycluster.tf` file:
+
+```tf
+module "controller" {
+  source = "git::https://github.com/kinvolk/lokomotive-kubernetes//kvm-libvirt/flatcar-linux/kubernetes?ref=<hash>"
+
+  providers = {
+    local    = "local.default"
+    null     = "null.default"
+    template = "template.default"
+    tls      = "tls.default"
+    libvirt  = "libvirt.default"
+  }
+
+  os_image_unpacked = "file:///home/CHANGEME/flatcar_production_qemu_image.img"
+
+  ssh_keys = [
+    "ssh-rsa AAAAB3Nza...",
+  ]
+
+  asset_dir = "/home/CHANGEME/clusters/asset"
+
+  machine_domain = "vmcluster.k8s"
+  cluster_name = "vmcluster"
+  node_ip_pool = "192.168.192.0/24"
+
+  controller_count = 1
+
+}
+
+module "worker-pool-one" {
+  source = "git::https://github.com/kinvolk/lokomotive-kubernetes//kvm-libvirt/flatcar-linux/kubernetes/workers?ref=<hash>"
+
+  providers = {
+    local    = "local.default"
+    template = "template.default"
+    tls      = "tls.default"
+    libvirt  = "libvirt.default"
+  }
+
+  ssh_keys = "${module.controller.ssh_keys}"
+
+  machine_domain = "${module.controller.machine_domain}"
+  cluster_name = "${module.controller.cluster_name}"
+  libvirtpool = "${module.controller.libvirtpool}"
+  libvirtbaseid = "${module.controller.libvirtbaseid}"
+
+  pool_name = "one"
+
+  count = 1
+
+  kubeconfig = "${module.controller.kubeconfig}"
+
+  labels = "node.supernova.io/role=backend,node-role.kubernetes.io/backend="
+}
+
+```
+
+Now download the Flatcar image (example here is for Flatcar Edge) and increase its size as desired for your use case:
+
+```sh
+$ wget https://edge.release.flatcar-linux.net/amd64-usr/2135.99.0/flatcar_production_qemu_image.img.bz2
```

Comment:
`wget`, `bunzip2` and `qemu-img` are not listed as requirements so I expect these commands to fail for some users.